common/inventory_parser/platform/c++17/inventory_parser.cpp

```diff
@@ -41,9 +41,8 @@ kvp::ExpectedKeyValuesMap GetInventoryData(const string &generators_dir) {
 		return kvp::ExpectedKeyValuesMap(data);
 	}
 
-	fs::directory_iterator end_itr; // default construction yields past-the-end
-	for (fs::directory_iterator itr(dir_path); itr != end_itr; ++itr) {
-		fs::path file_path = itr->path();
+	for (auto const &entry : fs::directory_iterator {dir_path}) {
```

Comment:
Hmm, I don't fully remember this, but with const used like this, should it not be :+1: 
`const auto const&` ?

Reply:
Why? The `entry` itself is not constant, you assign different values to it. And as we discussed before, we are not so strict with `const`...